aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StoragePathGetUrlTest.kt

```diff
@@ -79,4 +83,27 @@ class AWSS3StoragePathGetUrlTest {
         assertEquals("/public/$SMALL_FILE_NAME", result.url.path)
         assertTrue(result.url.query.contains("X-Amz-Expires=30"))
     }
+
+    @Test
+    fun testGetUrlWithObjectExistenceValidationEnabled() {
+        val result = synchronousStorage.getUrl(
+            SMALL_FILE_PATH,
+            AWSS3StorageGetPresignedUrlOptions.builder().setValidateObjectExistence(true).expires(30).build()
+        )
+
+        assertEquals("/public/$SMALL_FILE_NAME", result.url.path)
+        assertTrue(result.url.query.contains("X-Amz-Expires=30"))
+    }
+
+    @Test
+    fun testGetUrlWithStorageExceptionObjectNotFoundThrown() {
+        val exception = assertThrows(StorageException::class.java) {
+            synchronousStorage.getUrl(
+                StoragePath.fromString("public/SOME_UNKNOWN_FILE"),
+                AWSS3StorageGetPresignedUrlOptions.builder().setValidateObjectExistence(true).expires(30).build()
+            )
+        }
+
+        assertTrue(exception.cause is NotFound)
+    }
```

Comment:
Just thought of the case where `ValidateObjectExistence` is set to false (or not set at all) and the path is SOME_UNKNOWN_FILE, the expectation is that it doesn't throw an exception but when trying to resolve the retrieved URL, the response is not found. I don't think this case is written here yet